src/libraries/System.Private.CoreLib/src/System/IO/Strategies/Net5CompatFileStreamStrategy.Unix.cs

```diff
@@ -103,6 +104,28 @@ private void Init(FileMode mode, FileShare share, string originalPath, FileOptio
                     }
                 }
             }
+
+            // If allocationSize has been provided for a creatable and writeable file
+            if (allocationSize > 0 && (_access & FileAccess.Write) != 0 && mode != FileMode.Open && mode != FileMode.Append)
```

Comment:
This is only treating allocationSize as a hint that can be used if all of the above conditions are true, right?
Did you consider throwing on incorrect usage of allocationSize instead?

Reply:
Consider adding a helper method for all of these conditions since they are also executed in Windows impl of  CreateFileOpenHandle.
https://github.com/dotnet/runtime/blob/6875a9fc1619a82ecef4442454ed7731a27ed93d/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/FileStreamHelpers.Windows.cs#L75

Reply:
I got inspired by `NtCreateFile` which behaves like that. It's just a perf hint, which throws an exception only when we know for sure that there is not enough disk space|the file is too big for the given file system. I think that it's going to increase the adaptation amongst the users.

I am going to add the helper